### PR TITLE
feat(Hero): add noPaddingY and transparent background props

### PR DESCRIPTION
**Pull request by marcoskolodny**
Issue: [Link](https://jira.tid.es/browse/WEB-1782)

### CODE REVIEW

**Review by atabel — COMMENTED** [review]
Checking the example playroom in the jira ticket I've seen some problems in desktop viewport:

- I still see some extra horizontal paddings that shouldn't be there:
image

- There is a `minHeight` that we should remove when `noPaddingY` is `true`:
image
image

Also, please include some screenshot test that cover these cases: responsive layout nesting, `noPaddingY`, background none over inverse.

**Comment by marcoskolodny** [comment]
> Checking the example playroom in the jira ticket I've seen some problems in desktop viewport:
> 
>     * I still see some extra horizontal paddings that shouldn't be there:
>       image
> 
>     * There is a `minHeight` that we should remove when `noPaddingY` is `true`:
>       image
>       image
> 
> 
> Also, please include some screenshot test that cover these cases: responsive layout nesting, `noPaddingY`, background none over inverse.

@atabel
- removed custom logic for horizontal margin and replaced it with an InternalResponsiveLayout
- removed minHeight when noPaddingY prop is used
- added screenshots to cover scenarios related to noPaddingY and nested ResponsiveLayouts (the one about background none over inverse is already covered in the skins-story).

File: src/hero.tsx [hunk 1]
```diff
@@ -24,13 +25,17 @@ import type {DataAttributes, RendersElement, RendersNullableElement} from './uti
...
 const Layout = ({children, isInverse}: LayoutProps) => {
     return (
-        <ThemeVariant isInverse={isInverse}>
-            <div className={styles.layout}>{children}</div>
-        </ThemeVariant>
+        <InternalResponsiveLayout
+            isInverse={isInverse}
+            className={styles.layout}
+            innerDivClassName={styles.layout}
+            shouldExpandWhenNested="desktop"
```
Comment (atabel) [inline src/hero.tsx:34]: only in desktop? why?

Comment (marcoskolodny) [reply src/hero.tsx:34]: In mobile, the picture is rendered outside the responsive layout and the content has some margin relative to the picture because of this responsive layout (expected and intended).

Without expanding (now):

Screenshot from 2024-04-12 16-28-28

If we expand it (not correct):
Screenshot from 2024-04-12 16-29-17

File: src/__private_stories__/skin-components-story.tsx [hunk 2]
```diff
@@ -382,6 +383,19 @@ export const Default: StoryComponent<Args> = ({variant}) => {
                             />
                         </div>
                     </Inline>
+
+                    {/** Hero */}
```
Comment (pladaria) [inline src/__private_stories__/skin-components-story.tsx:387]: :thinking:

**Review by pladaria — APPROVED** [review]

**Review by aweell — COMMENTED** [review]
Slideshow arrows seem not positioned in the right place when nesting responsive layouts?

https://tinyurl.com/2bfhv62m

File: src/__screenshot_tests__/__image_snapshots__/hero-screenshot-test-tsx-hero-no-vertical-padding-1-snap.png
Comment (yceballost) [inline src/__screenshot_tests__/__image_snapshots__/hero-screenshot-test-tsx-hero-no-vertical-padding-1-snap.png:1]: Hero with noPadding and bullets looks broken :(

image


[Example](https://mistica-m3jn5uk7h-tuentisre.vercel.app/playroom#?code=N4Igxg9gJgpiBcIA8BlANgS1gZwBYQHcAdAOwAIyCMAXXAIQFc00ZrtSKMSA3GAJ2wxGzVtgC8wAGYBDNIIC%2BHMjRgBbccACCfPtICeAOkl8IqgBTAyLEgHNa8MgGYy8gDRkzAfXdYAHgEoyMQA%2BDyUKJAAJfghwijIAI2kwAGsbEwYSKDEiEFgZJmpcuIpVGCgMaQkS%2BKQASVVpGxga%2BLJsPjAckFxqagAHbHgAemHsCAZOmANM7H60aTwDSFVhgEYANgAGLd8ATh3hgH5JSZIaSZaQVvjF-pgwagAlaWoMCG7N%2BD3i8jb44bBGqKP5tXAwaRQTAkGASJAAFSaZGoenu3WSb14uWC0Uh0JgSGGiJswRB-zI-T4rBoLG6AAUqW9qCxfuSmbTgABtXLwmlXdw8vkAJlyAF1OX5RWT-jgwHwMP03hASN14bgMNhlJrpFZlTYyLL5Yr3uQqLQyDq0HrkTBfNRkRB2jAYGR8ARkerNQQIHwUuxrqD4rbqLo4XSFmAYPg0LA%2BGRAdK2gkGH1lXDGKmSAyMI0%2BHoyDIUjBgppHibCRnqMrs7m9KSasnMwAZLgpdMpqskFskFKuqmSboAYmx3ZSFY7ytH9cDFCgr2kmj68sbMA0lmoq7eUAcuXBJlyLkT8RwKSr-QAsuVKnSINgaCbuvKbL0DyQIHTIRVbABNOKAshKP4-ggoCpAgK4IC0Goq4IJyIDnhA3AatQ0h8CAooQVQUC0NgsGONsrhrFsQoACxSkAA)

**Comment by marcoskolodny** [comment]
> Slideshow arrows seem not positioned in the right place when nesting responsive layouts?
> 
> https://tinyurl.com/2bfhv62m

After discussing offline with Alex, we've identified this is an issue with Slideshow, not related to Hero at all. I've created an issue for this: https://jira.tid.es/browse/WEB-1829

File: src/__screenshot_tests__/__image_snapshots__/hero-screenshot-test-tsx-hero-no-vertical-padding-1-snap.png
Comment (marcoskolodny) [reply src/__screenshot_tests__/__image_snapshots__/hero-screenshot-test-tsx-hero-no-vertical-padding-1-snap.png:1]: Both components are working as expected. If I put the slideshow inside a box with padding and add a blue border in Hero, you can see that the Hero's border is correct (fits the height of the content):

MicrosoftTeams-image

The problem is that slideshow paints the bullets ON TOP of the content (without adding space in the bottom). In cases like Hero, if you decide to remove the bottom padding (using noPaddingY prop), it may not look good.

If, instead of Slideshow, you use a regular Carousel (in which the bullets are painted BELOW the content) and hero with noPaddingY, you will see that everything fits better:

image

**Review by aweell — APPROVED** [review]

**Review by yceballost — APPROVED** [review]

File: src/hero.tsx (hunk 1 above)
Comment (atabel) [reply src/hero.tsx:34]: I think you're wrong. See:

https://mistica-m3jn5uk7h-tuentisre.vercel.app/playroom#?code=N4Igxg9gJgpiBcIA8AFATjAbgSxgdwBUIIAbAZwAIyALCPAYVIjQGUxqYBbGFmEmMABdmAPgA6AOwoUkACRhoIk6dIBGAQzABrAOaKArhKgBeMSAkQJMM8pUWU6qFGwSdATVvTuz9ceCeVJABJTnUdaykVKLI0MFMQakFBAAcyeAB6dLIIfViYADpDMmSSdRp8yE50gEYANgAGeoAPAE5G9IB%2BADNciWxBXOsQAKiy5IFBACV1QWwIeLr4FptIqPTxVYoAXwCORxIXGD8kAjCKQQBPcfjNWcwhkXl9w6R0050RHc3kjFnBfni6Bgf34KyiIKO-hABH6oJAXyisDIYDQ2GSs0s8QI1GwlFxFHUFBIlh0FCRKLRGKkeH61AJRJJ5xgTUE5wgVBgMAotDw5xxlDwzC0ZDBKmZgjQvmAqFKYBgtBIsDQFHWCJUqn0SUx0oAQprhBJ0NhQmgLhQuuotDARABBIRzCSvPVaw2ok0XT4BDUugAyLi0x2dBr9Ei03IwXXiAGIzCIQ1onfrLPHPZsoDN1DakqjvTAyH5gEyyLMoPAKGYOIozNs1dIkVphMkALIwHwoCBkfoO%2BKonSJUXrWxISZ55KWTv3H3qC45VmYdSo9QSQTxFz3NBkB4BOQKJSbNSaXQGIzxCxWUVReyOZyuDz7ijebBSkaBEJhCJRT9UWLxRIpNKZNkuRyoUEjFKU5SVDUDTNG09SdD0aB9AMGAXl%2BBLFBM0wYgstRLGhawbF%2BtYqHsUAHFYxzvOcVxHGYtzYPcsZPORLxvGEqZfj8wKwnRIBAhCBHnLxfhmDC-xDCRdZ5hS6LdmJ-IUPihLEq4ZIyaicmWBQNKCHSKmMoI4pshyXI8ny%2BKCmgwpCeKkrHCgsryqQSoqpxn7egagZJq6xoLmaFpWra9qWImLpGu67lRJ5yb%2Bt5vr%2BuGMCRmYMYgHG-phcG-pRSo6aCJm2bYLm%2BbAIWRnFtgpblgku7VlsUnqWQDYQM2rZPu2nZUj22B9oIaGDpErwjsU46MTAU4zpqGyvECOD4EQpBkBsIAADQgHpXB5ggADaIA6qU%2BggAAuutNJQHpZC7QAzA0q0AOy1AAHKt1T1AATAALMdWxAA

There are 2 Heros in this playroom. One inside a ResponsiveLayout and one outside it, the one inside the it should expand and be aligned as the one outside

File: src/hero.tsx [hunk 3]
```diff
@@ -147,11 +166,13 @@ const Hero = React.forwardRef<HTMLDivElement, HeroProps>(
...
                         }}
-                        className={classnames(styles.container, styles.containerMobile)}
+                        className={classnames(styles.container, styles.containerMobile, {
+                            [styles.containerMinHeight]: !noPaddingY,
+                        })}
                     >
                         {media}
                         <Layout isInverse={isInverse}>
-                            <Box paddingTop={24} paddingBottom={isInsideSlideShow ? 48 : 24}>
+                            <Box paddingTop={24} paddingBottom={noPaddingY ? 0 : isInsideSlideShow ? 48 : 24}>
```
Comment (kydorn) [inline src/hero.tsx:175]: if I understand this correctly, if I use noPaddingY inside a slideshow, I would need to add the pageBullets outside (or reserve this space inside the "slide", correct? It is okay, but just to make sure.

**Review by kydorn — APPROVED** [review]

File: src/hero.tsx (hunk 3 above)
Comment (marcoskolodny) [reply src/hero.tsx:175]: Yes, that is correct :)

Comment (marcoskolodny) [reply src/hero.tsx]: Actually, giving it a second thought: I believe it's okay to add the bottom padding in case we put the Hero inside a slideshow (and set it to 0 otherwise if the noPaddingY prop is true). I've fixed this

File: src/hero.tsx (hunk 1 above)
Comment (marcoskolodny) [reply src/hero.tsx]: It should be fixed now :)

File: src/hero.tsx [hunk 4]
```diff
@@ -123,16 +129,29 @@ const BACKGROUND_COLOR = {
...
     brand: vars.colors.backgroundBrand,
     'brand-secondary': vars.colors.backgroundBrandSecondary,
+    none: 'transparent',
 };
 
 const Hero = React.forwardRef<HTMLDivElement, HeroProps>(
     (
-        {height, background = 'default', media, desktopMediaPosition = 'left', dataAttributes, ...rest},
+        {
+            height,
```
[the remark this answers is not in the document]
Comment (marcoskolodny) [reply src/hero.tsx:138]: I think we can do it in a different one, along with the refactor to try to avoid issues when using ssr
